src/ast/IterationCounter.cpp

```diff
@@ -0,0 +1,22 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+#include "ast/IterationCounter.h"
+#include <ostream>
+
+namespace souffle::ast {
+
+void IterationCounter::print(std::ostream& os) const {
+    os << "@iteration";
```

Comment:
Should the printer be as close as possible to the original Souffle input source code?  Maybe `__ITERATION__`.